### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by khanh-devos**
- [x] Install CanCanCan in your project.
- [x] Add a role column to the users table. Remember to use a migration for this.
- [x] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [x] For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [x] For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(user)
+    return unless user.present?
+
+    return unless user.admin?
+
+    can :destroy, Post, user: user
+    can :destroy, Comment, user: user
+
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
```
Comment (mikethreels) [inline app/models/ability.rb]: - [x] Right now, if the user is the owner of a post/comment they will not be able to destroy it, please make sure to adjust the method so the owner will be able to delete the post/comment as well.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 class PostsController < ApplicationController
-  before_action :set_user, only: %i[show index]
-  before_action :set_post, only: %i[show]
+  # before_action :authenticate_user!, only: %i[create destroy]
+
+  before_action :set_user, only: %i[show index destroy]
+  before_action :set_post, only: %i[show destroy]
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb:7]: - [x] for the post-authorisation you should use the cancancan gem according to the requirements.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job implementing the destroy actions

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by khanh-devos** [comment]
Thank you very much @mikethreels and @kit0-0,
I will fix it now.

**Review by divinecharlotte — APPROVED** [review]
# Hello @khanh-devos 😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**